### PR TITLE
feat: make central collector injectable

### PR DESCRIPTION
**Pull request by VinozzZ**
## Which problem is this PR solving?

The new CentralCollector should replace the old InMemoryCollector. 

## Short description of the changes

- Implements the rest of the collector functionalities except stress relief
- Makes sure CentralCollector has access to all necessary information when making a decision for a trace as well as 
sending a kept trace.
- Since the `ParentID` is not currently used for anything but determine whether a span is root span, I replaced the `ParentID` field with `IsRoot` field on span objects
- Adapted applicable existing Refinery tests to use the new CentralCollector.

### CODE REVIEW

File: collect/central_collector.go [hunk 1]
```diff
@@ -19,13 +22,38 @@ import (
...
+	TraceSendGotRoot        = "trace_send_got_root"
+	TraceSendExpired        = "trace_send_expired"
+	TraceSendEjectedFull    = "trace_send_ejected_full"
+	TraceSendEjectedMemsize = "trace_send_ejected_memsize"
+	TraceSendLateSpan       = "trace_send_late_span"
+)
+
 const (
 	// TODO: these should be configurable
 	cacheEjectBatchSize         = 100
```
Comment (VinozzZ) [inline collect/central_collector.go:52]: this is not needed anymore

File: centralstore/smartwrapper_test.go [hunk 2]
```diff
@@ -83,12 +69,25 @@ func getAndStartSmartWrapper(storetype string) (*SmartWrapper, func(), error) {
...
-		{Value: store},
 		{Value: sw},
 	}
 	g := inject.Graph{Logger: dummyLogger{}}
+
+	var store BasicStorer
+	switch storetype {
+	case "local":
+		store = &LocalRemoteStore{}
+	case "redis":
```
Comment (VinozzZ) [inline centralstore/smartwrapper_test.go:81]: add another case "real-redis"

File: centralstore/smartwrapper_test.go [hunk 3]
```diff
@@ -311,9 +311,8 @@ func TestSetTraceStatuses(t *testing.T) {
 
 	err = store.RecordMetrics(ctx)
 	require.NoError(t, err)
-	value, ok := store.Metrics.Get("redisstore_count_traces")
+	_, ok := store.Metrics.Get("redisstore_count_traces")
 	require.True(t, ok)
-	require.EqualValues(t, 0, value)
```
Comment (VinozzZ) [inline centralstore/smartwrapper_test.go:322]: capture this initial value and compare the result with the initial value

**Review by kentquirk — APPROVED** [review]
We've talked about this at length so I'm approving; we'll still have cleanup to do but it's just merging to a decision branch, so let's go.

**Pull request merged**